src/components/Popover/index.tsx

```diff
@@ -113,8 +117,20 @@ const Popover = React.forwardRef<HTMLDivElement, PopoverProps>(
     useCloseOnScroll(closeOnScrollRef, isActive, () => setIsActive(false));
     useDisableScroll(disableScrollRef, isActive);
 
+    const [triggerRef, triggerDimensions] = useMeasure();
+
     const isHover = interactionMode === 'hover';
     const isControlled = interactionMode === 'controlled';
+    const finalAlignment =
+      alignment === 'center-right'
+        ? 'start'
+        : alignment === 'center-left'
+        ? 'end'
+        : alignment;
+    const finalAlignOffset =
+      alignment === 'center-right' || alignment === 'center-left'
+        ? triggerDimensions.width / 2 + alignmentOffset
+        : alignmentOffset;
```

Comment:
What do you think about making a reusable hook for this (similar to `useDisableScroll`)?

Reply:
i don't think this is going to be used somewhere else, i'm also confused by 'hook', wouldn't this just be a function if we want to extract the functionality from here?

Reply:
maybe something like this?

```
export const usePopoverAlignment = (
  initialAlignment: PopoverAlignment = 'start',
  initialAlignmentOffset = 0
): [ReturnType<typeof useMeasure>[0], RadixPopoverAlignment, number] => {
  const [ref, dimensions] = useMeasure();

  const alignment =
    initialAlignment === 'center-right'
      ? 'start'
      : initialAlignment === 'center-left'
      ? 'end'
      : initialAlignment;

  const alignmentOffset =
    initialAlignment === 'center-right' || initialAlignment === 'center-left'
      ? dimensions.width / 2 + initialAlignmentOffset
      : initialAlignmentOffset;

  return [ref, alignment, alignmentOffset];
};
```

// Popover/index.tsx
```
    // const [triggerRef, triggerDimensions] = useMeasure();
    // const finalAlignment =
    //   alignment === 'center-right'
    //     ? 'start'
    //     : alignment === 'center-left'
    //     ? 'end'
    //     : alignment;
    // const finalAlignmentOffset =
    //   alignment === 'center-right' || alignment === 'center-left'
    //     ? triggerDimensions.width / 2 + alignmentOffset
    //     : alignmentOffset;

    const [triggerRef, finalAlignment, finalAlignmentOffset] = usePopoverAlignment(
      alignment,
      alignmentOffset
    );
```
what do you all think?

Reply:
it cleans up our code but also seems very specific to this one use case

Reply:
I like the idea. Even if only used in this component, creating small functions makes it easier to understand the code and, with a good name, serves as documentation.